File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    int total = 0;
    float change;
    do
    {
        printf("How much change is owed? ");
        change = get_float();
    }
    while (change < 0);
    int cents = round(change * 100);
    while (cents >= quarter)
    {
        cents = cents - quarter;
        total++;
    }
    while (cents >= dime)
    {
        cents = cents - dime;
        total++;
    }
    while (cents >= nickel)
    {
        cents = cents - nickel;
        total++;
    }
    while (cents >= penny)
    {
        cents = cents - penny;
        total++;
    }
    printf("%i\n", total);
}

File: pset6/greedy.py
import cs50

# Main function
def main():
    # prompt for input
    while True:
        print("How much change is owed? ", end="")
        change = cs50.get_float();
        if change >= 0:
            break

    # declare variables
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1
    total = 0

    # calculate change
    cents = change * 100
    while cents >= quarter:
        cents = cents - quarter
        total += 1
    while cents >= dime:
        cents = cents - dime
        total += 1
    while cents >= nickel:
        cents = cents - nickel
        total += 1
    while cents >= penny:
        cents = cents - penny
        total += 1

    # return output
    print("{}".format(total))


if __name__ == "__main__":
    main()